x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/pipeline_processors_editor_item/pipeline_processors_editor_item.tsx

```diff
@@ -25,127 +34,147 @@ export interface Handlers {
 
 export interface Props {
   processor: ProcessorInternal;
-  selected: boolean;
   handlers: Handlers;
   selector: ProcessorSelector;
   description?: string;
+  movingProcessor?: ProcessorInfo;
+  renderOnFailureHandlers?: () => React.ReactNode;
 }
 
 export const PipelineProcessorsEditorItem: FunctionComponent<Props> = memo(
-  ({ processor, description, handlers: { onCancelMove, onMove }, selector, selected }) => {
+  ({
+    processor,
+    description,
+    handlers: { onCancelMove, onMove },
+    selector,
+    movingProcessor,
+    renderOnFailureHandlers,
+  }) => {
     const {
       state: { editor, processorsDispatch },
     } = usePipelineProcessorsContext();
 
     const disabled = editor.mode.id !== 'idle';
-    const isDarkBold =
-      editor.mode.id !== 'editingProcessor' || processor.id === editor.mode.arg.processor.id;
+    const moving = editor.mode.id === 'movingProcessor';
+    const selected = processor.id === movingProcessor?.id;
+    const isBeingEdited =
+      editor.mode.id === 'editingProcessor' && processor.id === editor.mode.arg.processor.id;
+    const isDimmed =
```

Comment:
I'd find this a bit easier to understand if we split this up into two variables and consumed them the same way `selected` and `isBeingEdited` are consumed on line 67. For example:

```ts
const isEditingOtherProcessor = editor.mode.id === 'editingProcessor' && !isBeingEdited;
const isMovingOtherProcessor = editor.mode.id === 'movingProcessor' && !selected;

const panelClasses = classNames({
  'pipelineProcessorsEditor__item--selected': selected || isBeingEdited,
  'pipelineProcessorsEditor__item--dimmed': isEditingOtherProcessor || isMovingOtherProcessor,
});
```

We could also adjust the other variable names for further symmetry: `isMovingThisProcessor`, `isEditingThisProcessor`, `isEditingOtherProcessor`, `isMovingOtherProcessor`.